services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/api_keys_manager.py

```diff
@@ -71,97 +81,138 @@ async def is_used(self, identifier: str, cleanup_context: CleanupContext) -> boo
     async def _create(  # type:ignore [override] # pylint:disable=arguments-differ
         self, identifier: str, product_name: ProductName, user_id: UserID
     ) -> tuple[str, ApiKeyGet]:
-        result = await self.rpc_client.request(
-            WEBSERVER_RPC_NAMESPACE,
-            parse_obj_as(RPCMethodName, "create_api_keys"),
+        result = await create_api_key_and_secret(
+            self.app,
             product_name=product_name,
             user_id=user_id,
-            new=ApiKeyCreate(display_name=identifier, expiration=None),
+            name=identifier,
+            expiration=None,
         )
         return identifier, ApiKeyGet.parse_obj(result)
 
     async def get(  # type:ignore [override] # pylint:disable=arguments-differ
         self, identifier: str, product_name: ProductName, user_id: UserID
     ) -> ApiKeyGet | None:
-        result: Any | None = await self.rpc_client.request(
-            WEBSERVER_RPC_NAMESPACE,
-            parse_obj_as(RPCMethodName, "api_key_get"),
-            product_name=product_name,
-            user_id=user_id,
-            name=identifier,
+        return await get_api_key_and_secret(
+            self.app, product_name=product_name, user_id=user_id, name=identifier
         )
-        return parse_obj_as(ApiKeyGet | None, result)
 
-    async def _destroy(self, identifier: str, cleanup_context: CleanupContext) -> None:
-        await self.rpc_client.request(
-            WEBSERVER_RPC_NAMESPACE,
-            parse_obj_as(RPCMethodName, "delete_api_keys"),
+    async def _destroy(self, identifier: str, cleanup_context: _CleanupContext) -> None:
+        await delete_api_key_and_secret(
+            self.app,
             product_name=cleanup_context.product_name,
             user_id=cleanup_context.user_id,
             name=identifier,
         )
 
 
-async def get_or_create_api_key(
+def _get_identifier(node_id: NodeID, run_id: RunID) -> str:
+    # Generates a new unique key name for each service run
+    # This avoids race conditions if the service is starting and
+    # an the cleanup job is removing the key from an old run which
+    # was wrongly shut down
+    return f"_auto_{uuid5(node_id, run_id)}"
+
+
+async def _get_or_create(
     app: FastAPI,
     *,
     product_name: ProductName,
     user_id: UserID,
     node_id: NodeID,
     run_id: RunID,
 ) -> ApiKeyGet:
-    api_keys_manager = _get_api_keys_manager(app)
-    display_name = _get_api_key_name(node_id, run_id)
+
+    warnings.warn(
```

Comment:
waht is this for?

Reply:
It's a deprecation warning. The new interface is in a different module, as described in the deprecation error

Reply:
all good missed the message